lib/util.js

```diff
@@ -63,11 +63,15 @@ function hashObject(labels) {
 		keys = keys.sort(); // need consistency across calls
 	}
 
-	const elems = [];
-	for (let i = 0; i < keys.length; i++) {
-		elems.push(`${keys[i]}:${labels[keys[i]]}`);
+	let hash = '';
+	const size = keys.length;
+	for (let i = 0; i < size; i++) {
+		hash += `${keys[i]}:${labels[keys[i]]}`;
+		if (i !== size - 1) {
+			hash += ',';
+		}
```

Comment:
I haven't benchmarked here, but in other TSV generators, the fastest approach I found (vs. `vals.join(",")`, `if (not last element) str += ","` or `str.substring(0, str.length - 1)`) was:

```js
for (let i = 0; i < size - 1; i++) {
  hash += `${keys[i]}:${labels[keys[i]]},`; // <-- comma here
}
hash += `${keys[i]}:${labels[keys[i]]}`;
```

---

Other than the lint failure (line endings, it looks like), looks good to me!

Reply:
That is a good idea. I'll benchmark it.

The lint failure seams unrelated to me. The same failure occurs on the master branch, too:
https://ci.appveyor.com/project/siimon/prom-client/builds/21269249

Reply:
Your suggestion is indeed faster:

```bash
### Summary:

✓ registry ➭ getMetricsAsJSON#1 with 64 is 6.030% faster.
⚠ registry ➭ getMetricsAsJSON#2 with 8 is 4.290% acceptably slower.
⚠ registry ➭ getMetricsAsJSON#2 with 4 and 2 with 2 is 0.9444% acceptably slower.
✓ registry ➭ getMetricsAsJSON#2 with 2 and 2 with 4 is 2.637% faster.
⚠ registry ➭ getMetricsAsJSON#6 with 2 is 0.7704% acceptably slower.
✓ registry ➭ metrics#1 with 64 is 10.78% faster.
✓ registry ➭ metrics#2 with 8 is 12.85% faster.
✓ registry ➭ metrics#2 with 4 and 2 with 2 is 8.681% faster.
✓ registry ➭ metrics#2 with 2 and 2 with 4 is 7.691% faster.
✓ registry ➭ metrics#6 with 2 is 5.984% faster.
✓ histogram ➭ observe#1 with 64 is 6.824% faster.
✓ histogram ➭ observe#2 with 8 is 37.59% faster.
✓ histogram ➭ observe#2 with 4 and 2 with 2 is 33.50% faster.
✓ histogram ➭ observe#2 with 2 and 2 with 4 is 31.08% faster.
✓ histogram ➭ observe#6 with 2 is 26.26% faster.
✓ summary ➭ observe#1 with 64 is 13.52% faster.
✓ summary ➭ observe#2 with 8 is 37.20% faster.
✓ summary ➭ observe#2 with 4 and 2 with 2 is 32.39% faster.
✓ summary ➭ observe#2 with 2 and 2 with 4 is 33.90% faster.
✓ summary ➭ observe#6 with 2 is 31.74% faster.
```